client/src/components/MenuItempage.jsx

```diff
@@ -105,12 +135,28 @@ class MenuItempage extends React.Component {
     const { id } = this.props.location.state;
     axios.get(`/api/reviews`)
       .then((resp) => {
-        // filter for reviews of the current menu item only
-        const reviews = resp.data.filter(review => 
-          review.menuItem === id);
-        const itemReviews = reviews.map(review =>  
+        /**
+         * Filter for reviews of the current menu item only.
+         * Only reviews with texts are displayed, however
+         * rating only reviews (from extension) from the extension are
+         * factored in for the calculation of the rating
+         */
+        var reviews = resp.data.filter(review => 
+          review.menuItem === id && review.comments != null);
+        if (this.state.loggedIn) {
+          const userReview = resp.data.filter(review => review.author === this.state.user._id && review.menuItem === id);
+          if (userReview.length !== 0) {
+            this.setState({alreadyRated: true});
```

Comment:
does this also mean that we're verifying if alreadyRated is true every single time around? ideally if alreadyRated is true, it will stay true so that we only need to do this once

Reply:
This only happens once when the function getReviews() is called, which only is called upon mounting and updating the page after submitting a review